Jenkinsfile

```diff
@@ -1,136 +1,148 @@
-                    sh './docker/env/hydra.sh unit-tests'
-                    if (env.CHANGE_ID) {
-                        pullRequest.createStatus(status: 'success',
-                                         context: 'jenkins/unittests',
-                                         description: 'All unit tests passed',
-                                         targetUrl: "${env.JOB_URL}/workflow-stage")
-                    }
-                } catch(Exception ex) {
-                    if (env.CHANGE_ID) {
-                        pullRequest.createStatus(status: 'failure',
-                                     context: 'jenkins/unittests',
-                                     description: 'unit tests failed',
-                                     targetUrl: "${env.JOB_URL}/workflow-stage")
-                    }
-                    currentBuild.result = 'UNSTABLE'
-                }
-            }
+      steps {
+        script {
+          try {
+            sh './docker/env/hydra.sh unit-tests'
+            markPullRequest('success', 'jenkins/unittests', 'All unit tests are passed')
+          } catch(Exception ex) {
+            markPullRequest('failure', 'jenkins/unittests', 'Some unit tests failed')
+          }
         }
+      }
     }
     stage("lint test-cases") {
-        steps {
-            script {
-                try {
-                    sh ''' ./utils/lint_test_cases.sh '''
-
-                    if (env.CHANGE_ID) {
-                        pullRequest.createStatus(status: 'success',
-                                         context: 'jenkins/lint_test_cases',
-                                         description: 'all test cases are checked',
-                                         targetUrl: "${env.JOB_URL}/workflow-stage")
-                    }
-                } catch(Exception ex) {
-                    if (env.CHANGE_ID) {
-                        pullRequest.createStatus(status: 'failure',
-                                         context: 'jenkins/lint_test_cases',
-                                         description: 'some test cases failed to check',
-                                         targetUrl: "${env.JOB_URL}/workflow-stage")
-                    }
-                    currentBuild.result = 'UNSTABLE'
-                }
-            }
+      steps {
+        script {
+          try {
+            sh ''' ./utils/lint_test_cases.sh '''
+            markPullRequest('success', 'jenkins/lint_test_cases', 'All test cases are passed')
+          } catch(Exception ex) {
+            markPullRequest('failure', 'jenkins/lint_test_cases', 'Some test cases failed')
+          }
         }
+      }
     }
     stage("provision test") {
-        when { changeRequest() }
-        steps {
+      when {
+        expression {
+          return lookForLabels("test-provision,test-provision-aws,test-provision-gce") && currentBuild.result == null
+        }
+      }
+      failFast true
+      parallel {
```

Comment:
:+1: for using parallel here, I would also add the docker backend when it's ready

Reply:
We will add, but we will need to keep docker host somewhere in the cloud for these purposes.